src/components/ListItem.jsx

```diff
@@ -1,9 +1,46 @@
 import './ListItem.css';
+import { useEffect } from 'react';
+import { updateItem } from '../api/firebase';
+
+export function ListItem({ listToken, item }) {
+	const millisecondsIn24hrs = 84600000;
+	useEffect(() => {
+		if (item.isChecked) {
+			let calcDate =
+				item.dateLastPurchased.seconds * 1000 + millisecondsIn24hrs;
+
+			setTimeout(() => {
```

Comment:
this setTimeout would only work if you kept the app open for 24hrs. The point is that it updates everytime. you can achieve that  by checking if 24hrs have passed from `dateLastPurchased` :) Also since the `updateItem` is an async function you need to make the function in useEffect async as well and await the updateItem

Reply:
This is resolved